pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -72,6 +76,25 @@ func (p *ConfigurationEditFlags) addSharedFlags(command *cobra.Command) {
 			"any number of times to set multiple environment variables. "+
 			"To unset, specify the environment variable name followed by a \"-\" (e.g., NAME-).")
 	p.markFlagMakesRevision("env")
+
+	command.Flags().StringArrayVarP(&p.EnvFrom, "env-from", "", []string{},
+		"Config a envfrom with a config map or secret. (config-map | secret):CONFIG_MAP_OR_SECRET_NAME "+
+			"you may provide this flag any number of times. "+
+			"To unset, specify the config map name followed by a \"-\" (e.g., config-map:CONFIG_MAP_NAME- or secret:SECRET_NAME-).")
+	p.markFlagMakesRevision("env-from")
+
+	command.Flags().StringArrayVarP(&p.VolumeMount, "volume-mount", "", []string{},
+		"Config a volume mount. /mount/path=VOLUME_MOUNT ; "+
+			"you may provide this flag any number of times to mount multiple volumes. "+
+			"To unset, specify the mount path followed by a \"-\" (e.g., /mount/path-).")
+	p.markFlagMakesRevision("volume-mount")
+
+	command.Flags().StringArrayVarP(&p.Volume, "volume", "", []string{},
+		"Config a volume with a config map or secret. VOLUME_NAME=(config-map|secret):CONFIG_MAP_OR_SECRET_NAME ; "+
+			"you may provide this flag any number of times to define multiple volumes. "+
+			"To unset, specify the volume name followed by a \"-\" (e.g., /mount/path-).")
+	p.markFlagMakesRevision("volume")
```

Comment:
My original proposal was using the combined flag as you suggested. However, as you can see in https://github.com/knative/client/pull/393#discussion_r320467050, there was other opinion about giving more freedom to user for mounting a volume.

Basically, I think that, for this kind of CLI, it is very important to provide all the functionalities without assumptions as well as to provide convenience.  So, I would like to suggest to provide all the flags: `--volume-mount /data=mydata --volume mydata=cm:myconfigmap` and `--mount /data=mydata:cm:myconfigmap`.  How do you think about it?

Reply:
@igsong thanks for the comment reference, I missed that. However, if I understand @sixolet correctly, she also opts for ease of use over completeness and covering every corner case.

I agree, if we can support both, a simple, single option use case with creating both entries (volume + mount) in one go and for more advanced uses case a second option to allow volumes to be reused, then I'm ok (but as said, we definitely should make it as easy as possible to use volumes). 

I like your suggestion, but I wonder whether we can make it even easier for the user as the volume name itself is actually an implementation detail (the user never will see this name again when using `kn` only). So I like the idea to have it autogenerated (defaulting to the cm/secret name, adding suffixes if not unique).

So my suggestion would be for the mount option, having multiple parts seperate by `:`:

* 1 part ("mydata") --> consider it a volume name
* 2 parts ("cm:mycm", "secret:mysecret") -> mount the cm or secret and create a volume name implicitely (here: configmap with name "mycm" creates a volume with name "mycm")
* 3 parts ("cm:mycm:mydata") --> mount a configmap and create a volume of the given name (here: configmap "mycm" is mounted by creating a volume "mydata")

wdyt ?

Reply:
Thank you for comments. I agree with that most of users don't need to specify an explicit volume name, but, if we provide 3 kinds of data format for one flag and they are distinguished just by `:` , it may cause confusion or looks like too complicated. So my refined suggestion would be following.

* When the user want to mount a existing volume, `--mount /mydir=volume:myvolume`
* When the user want to mount a configmap or secret(a volume is automatically generated),  `--mount /mydir=(cm:myconfigmap|secret:mysecret)`
* When the user want to define a volume explicitly and mount the volume `--volume myvolume=(cm:myconfigmap|secret:mysecret) --mount /mydir=volume:myvolume`

Reply:
Sounds good to me, with maybe the refined to omit the `volume:` prefix for volumes, and use a prefix only for `cm` (`configmap`) and `secret`. I think this is still unique and imo easier to understand. (as you would use the volume name only in `--volume myvolume=cm:myconfig` instead of `--volume volume:myvolume=cm:myconfig`)

Reply:
OK, I've changed as you mentioned.